docs/madr/decisions/011-mesh-traffic-route.md

```diff
@@ -0,0 +1,555 @@
+            prefix: /
+        filters:
+        - requestHeader:
+            add:
+            - name: X-Client
+              value:
+        backendRefs:
+        - weight: 100
+          kind: MeshServiceSubset
+          name: svc_a
+          tags:
+            version: v1
+```
+
+But the service owner probably expects to somehow
+be able to "enforce" routes on consumers that can't be overriden.
+
+Perhaps we finally need [`override`](https://gateway-api.sigs.k8s.io/v1alpha2/references/policy-attachment/#hierarchy)?
+
+##### Merging
+
+Is there a use case for being able to write a policy that combines with the above `owner` policy such that the final resource would have both `filters`?
+
+Perhaps given the above `owner` policy along with a different `svc_a` specific
+policy:
+
+```yaml
+metadata:
+  name: consumer
+spec:
+ targetRef:
+   kind: MeshService
+   name: svc_b
+ to:
+  - targetRef:
+      kind: MeshService
+      name: svc_a
+    default:
+      rules:
+      - matches:
+          path:
+            prefix: /
+        filters:
+        - requestHeader:
+            add:
+            - name: X-Client
+              value:
+```
+
+a user might expect:
+
+```yaml
+spec:
+ targetRef:
+   kind: MeshService
+   name: svc_b
+ to:
+  - targetRef:
+      kind: MeshService
+      name: svc_a
+    default:
+      rules:
+      - matches:
+          path:
+            prefix: /
+        filters:
+        - requestHeader:
+            add:
+            - name: X-Client
+              value:
+        - responseHeader:
+            add:
+            - name: X-Some-Header
+              value: something
+        backendRefs:
+        - weight: 100
+          kind: MeshServiceSubset
+          name: svc_a
+          tags:
+            version: v1
+```
+
+Could we achieve this by always combining `filters` using `matches` as a merge key
+with structural equality and
+specifying `default`/`override` and maybe even `merge` for `filters` _next to_
+`matches`?
+It probably doesn't make sense to merge `backendRefs` so they should be
+limited to `default`/`override`.
+
+Given:
+
+```yaml
+metadata:
+  name: owner
+spec:
+ targetRef:
+   kind: Mesh
+ to:
+  - targetRef:
+      kind: MeshService
+      name: svc_a
+    rules:
+    - matches:
+        path:
+          prefix: /
+      merge:
+        filters:
+        - responseHeader:
+            add:
+            - name: X-Some-Header
+              value: something
+      override:
+        backendRefs:
+        - weight: 90
+          kind: MeshServiceSubset
+          name: svc_a
+          tags:
+            version: v1
+        - weight: 10
+          kind: MeshServiceSubset
+          name: svc_a
+          tags:
+            version: v2
+```
+
+along with:
+
+```yaml
+metadata:
+  name: owner
+spec:
+ targetRef:
+   kind: MeshService
+   name: svc_b
+ to:
+  - targetRef:
+      kind: MeshService
+      name: svc_a
+    rules:
+    - matches:
+        path:
+          prefix: /
+      merge:
```

Comment:
How do we handle a situation that the service owner defines `merge` because wants a specific header to be added to all requests but a more specific policy does `override`? Should we override?

Reply:
[A more specific policy can't `override` a less specific one](https://gateway-api.sigs.k8s.io/v1alpha2/references/policy-attachment/#hierarchy). In this example the `merge` from the less specific policy acts as a sort of mergeable `override`, as opposed to `override` which is unchangeable. See the above question from Jakub though where I think your question is also addressed.

Reply:
I.e. `override`/`default` refers I think to how a policy affects more specific policies after it rather than how a policy affects the less specific policies before it.